### PR TITLE
stat: add matrix conditioning functions for MCA

### PR DESCRIPTION
**Pull request by kortschak**
Replaces #449.

WIP - needs tests.

### CODE REVIEW

File: stat/pca_helpers.go [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+)
+
+// TransformDisjunctive transforms a Complete Disjunctive Table to a Transformed
+// Complete Disjunctive Table in order to be able to perform Multiple
+// Correspondence Analysis using PC.PrincipalComponents. TransformDisjunctive
+// places the TCDT in dst and returns it.
+//
+// If dst is nil, a new mat.Dense is allocated. If dst is not a zero matrix,
+// the dimensions of dst and cdt must match otherwise TransformDisjunctive will
+// panic. If cdt contains values other 0 or 1 TransformDisjunctive will panic.
```
Comment (vladimir-ch) [inline stat/pca_helpers.go]: s/other 0/other than 0/ ?

Comment (kortschak) [reply stat/pca_helpers.go]: Fixed

**Comment by btracey** [comment]
The code looks good. Is there a link to somewhere to find what this does? Even Googling left me unable to find more information.

**Comment by kortschak** [comment]
I'm waiting for the OP to come and comment. I think I got the approach from the [WP article](https://en.wikipedia.org/wiki/Multiple_correspondence_analysis).

**Comment by nathanhack** [comment]
@btracey and @kortschak sorry for the delayed response.  

The point/benefit of MCA is that it enables analysis of qualitative data sets.  The TransformDisjunctive function in this commit is the second step (in MCA) required to transform a **qualitative** data set so that it can be analyzed by PCA (**quantitative** analysis tool).  

Many attempts to use PCA on qualitative values over the years have shown that without this change the analysis results are skewed which can yield incorrect assumptions/results.

**Comment by kortschak** [comment]
@nathanhack, please understand that this is a low priority for me, so without more significant input this is likely to languish.

**Comment by kortschak** [comment]
I have had a look at the links posted above; the FactoMineR code is not well aligned with the Gonum approach and will require significant investment in time to be able to untangle, and neither link is particularly helpful for developing tests for an implementation.

In the absence of any significant input here, I'm going to abandon this work. I will leave the branch so it can be picked up again if the interest does actually present itself.

**Pull request closed without merging**